### ISSUE TITLE
Tune uWSGI settings based on application profile

### ISSUE THREAD

**Issue by rhysyngsun** [mitodl/mitxonline#1682]
See my writeup on how to approach this: https://github.com/mitodl/hq/discussions/393

The current settings on xPro (workers = 3, threads = 25) is probably a good starting point for values.

NOTE: MicroMasters is an older application, it's probably going to be more difficult to do this work, so I would recommend not spending a lot of time fine-tuning the settings and instead apply some reasonable settings, monitor on newrelic, and tune further if needed.

**Comment by pdpinch** [comment]
> Creating the locustfile.py for this - Ibrahim did not add an example file to xpro that I saw, but I plan to both to aid in quicker testing and as a breadcrumb for future tests

@ibrahimjaved12 was working on ocw-studio. If he hasn't shared the file in that repo, he should. 

Otherwise, this looks like the right plan. Let's get started!

**Comment by JenniWhitman** [comment]
I meant OCW :facepalm: The locust file is in a comment on the PR, so I have his structure (thankfully! Hence why I expect this to be wrapped up so quickly), but i think the better breadcrumb moving forward would be to include things in a doc or example file. We have a docs folder on mitxonline, so I'm planning to make sure it's noted so we can "re-tune" quickly and easily in the future :)

**Comment by JenniWhitman** [comment]
This work helped identify performance issues on /catalog of the redesign.  That is being tracked here: https://github.com/mitodl/hq/issues/2781. The completion of this work is pending changes to the catalog page and completion of that ticket. At that point I will rerun tests and we'll work with what we have then.

**Comment by pdpinch** [comment]
@JenniWhitman are you actively working on this? If not, we should move it back to in triage.

**Comment by JenniWhitman** [comment]
@pdpinch 
No, I'm not actively working on it right now, I was waiting on completion of the catalog work, however, seeing as we're not making any more major API changes we COULD go back and rerun this/do some tuning.

**Comment by JenniWhitman** [comment]
(so, ready column over triage potentially)
